src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/serverprocess.cpp

```diff
@@ -1118,6 +1125,52 @@ SERVER_PROCESS::SetupStdHandles(
     return hr;
 }
 
+
+void
+SERVER_PROCESS::ReadStdErrHandle(
+    LPVOID pContext
+)
+{
+    auto pLoggingProvider = static_cast<SERVER_PROCESS*>(pContext);
+    DBG_ASSERT(pLoggingProvider != NULL);
+    pLoggingProvider->ReadStdErrHandleInternal();
+}
+
+void
+SERVER_PROCESS::ReadStdErrHandleInternal()
+{
+    std::string tempBuffer;
+    tempBuffer.resize(4096);
+    size_t m_charactersLeft = 30000;
+
+    // If ReadFile ever returns false, exit the thread
+    DWORD dwNumBytesRead = 0;
+    while (true)
+    {
+        if (ReadFile(m_hStdoutHandle,
+            tempBuffer.data(),
+            4096,
+            &dwNumBytesRead,
+            nullptr))
+        {
+            auto text = to_wide_string(tempBuffer.substr(0, dwNumBytesRead), GetConsoleOutputCP());
```

Comment:
super nit: so many allocations here, could make a `to_wide_string` function that takes the string and count so you don't need to `substr`